pkg/controller/machineset/machineset_controller.go

```diff
@@ -494,12 +494,9 @@ func (r *ReconcileMachineSet) waitForMachineDeletion(machineList []*clusterv1.Ma
 func (r *ReconcileMachineSet) MachineToMachineSets(o handler.MapObject) []reconcile.Request {
 	result := []reconcile.Request{}
 
-	m := &clusterv1.Machine{}
-	key := client.ObjectKey{Namespace: o.Meta.GetNamespace(), Name: o.Meta.GetName()}
-	if err := r.Client.Get(context.Background(), key, m); err != nil {
-		if !apierrors.IsNotFound(err) {
-			klog.Errorf("Unable to retrieve Machine %q for possible MachineSet adoption: %v", key, err)
-		}
+	m, ok := o.Object.(*clusterv1.Machine)
+	if !ok {
+		klog.Errorf("failed convert object to machine %v", o.Object)
```

Comment:
I frequently see `%T` used to print out the type of object. I'd recommend doing that, and if you want `%v` or `%#v` as well, you could do that too.

Reply:
Something like `expected a Machine but got a %T`